server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/ElasticSearchMailboxModule.java

```diff
@@ -137,12 +137,12 @@ private ElasticSearchMailboxConfiguration getElasticSearchMailboxConfiguration(P
     @Singleton
     protected Client provideClient(ElasticSearchConfiguration configuration,
                                    ElasticSearchMailboxConfiguration mailboxConfiguration,
-                                   ElasticSearchQuotaConfiguration quotaConfiguration,
-                                   AsyncRetryExecutor executor) throws ExecutionException, InterruptedException {
+                                   ElasticSearchQuotaConfiguration quotaConfiguration) {
 
-        return RetryExecutorUtil.retryOnExceptions(executor, configuration.getMaxRetries(), configuration.getMinDelay(), NoNodeAvailableException.class)
-            .getWithRetry(context -> connectToCluster(configuration, mailboxConfiguration, quotaConfiguration))
-            .get();
+        return Mono.fromCallable(() -> connectToCluster(configuration, mailboxConfiguration, quotaConfiguration))
+            .retryBackoff(configuration.getMaxRetries(), Duration.ofMillis(configuration.getMinDelay()))
```

Comment:
I'm not sure a backoff strategy makes sense for the connection retry

Reply:
Can you develop a bit more? AsyncRetry stuff was providing a retry isn't it?